### DISCUSSION TITLE
[F# / C# interop] record types in both language is unable to be used in another

### DISCUSSION THREAD

**Discussion by Xyncgas** [dotnet/roslyn#66323] [General]
for example, in F# you can use
```F#
[<Struct; StructLayout(LayoutKind.Explicit)>]
type Data =
    {
                [<FieldOffsetAttribute(0)>]
                Memory : Vector4
                [<FieldOffsetAttribute(0)>]
                First: float32
    }

let X ={ Memory = Vector4.Zero; First = 0f }

let k = { X with First = 1f }
```

In C# you can use
```C#
public record class Data
{
    public Vector4 Memory;
    public float First;
}

var X = Data();

var k = X with { First = 1f };
```

But you can't use these records in the other language with their respective syntax for reassigning values to these records (in visual studio they don't show up as records too from syntax highlight)

One alternative I've been using is simply writing a duplicate structure in the language I am using, convert it to the underlying data structure from another language when I am done working

So the compiler can do this, where if the record type is in another language, generate a record type in this language. Whatever is incompatible, solve it in a intermediary layer, therefore it's transparent to the user and actually makes it usuable

**Comment by saint4eva** [comment]
Maybe, if you new up your Data class properly, it could show up
